src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -2453,6 +2491,106 @@ public override Task<ChildWorkflowHandle<TWorkflow, TResult>> StartChildWorkflow
                 return handleSource.Task;
             }
 
+            /// <inheritdoc/>
+            public override Task<NexusOperationHandle<TResult>> StartNexusOperationAsync<TResult>(
+                StartNexusOperationInput input)
+            {
+                var token = input.Options.CancellationToken ?? instance.CancellationToken;
+                // We do not even want to schedule if the cancellation token is already cancelled.
+                // We choose to use cancelled failure instead of wrapping in Nexus failure which is
+                // similar to what Java and TypeScript do, with the accepted tradeoff that it makes
```

Comment:
```suggestion
                // similar to what TypeScript do, with the accepted tradeoff that it makes
```
Java does wrap the cancelled failure in a Nexus failure. What does dotnet do in other immediate cancellation scenarios ?

Reply:
To clarify, this is copied from the child workflow comment. I was under the impression for child/nexus cancel-before-start, that Java and TypeScript just throw (i.e. set promise result to) a `CanceledFailure`, not a wrapped cancel failure as would happen if the cancel happened after schedule/start.

Reply:
>I was under the impression for child/nexus cancel-before-start, that Java and TypeScript just throw (i.e. set promise result to) a CanceledFailure

That is not true for Java, cancel-before-start throws the same error chain as cancel happened after schedule/start.